### PR TITLE
Improvements to testing, benchmarking, and tooling

### PR DESCRIPTION
**Pull request by ErikBjare**
- Added Pipfile
 - Added use of pytest-flask
 - Added use of pytest-benchmark
 - Refactored creation of `Flask`/`app` object into `create_app` function

### CODE REVIEW

File: aw_server/rest.py [hunk 1]
```diff
@@ -112,21 +110,22 @@ class BucketsResource(Resource):
...
     def get(self, bucket_id):
-        return app.api.get_bucket_metadata(bucket_id)
+        return current_app.api.get_bucket_metadata(bucket_id)
 
     @api.expect(create_bucket)
     @copy_doc(ServerAPI.create_bucket)
     def post(self, bucket_id):
         data = request.get_json()
-        bucket_created = app.api.create_bucket(bucket_id, event_type=data["type"], client=data["client"], hostname=data["hostname"])
+        print(data)
```
Comment (ErikBjare) [inline aw_server/rest.py]: Oops, this should be removed.

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+aw-core = {ref = "master", git = "https://github.com/ActivityWatch/aw-core.git"}
+flask-restplus = ">=0.9.2"
+pymongo = ">=3.2"
```
Comment (ErikBjare) [inline Pipfile]: This does not belong here.

File: Makefile [hunk 3]
```diff
@@ -18,11 +18,16 @@ install:
 	cp misc/aw-server.service /usr/lib/systemd/user/aw-server.service
 
 test:
-	python3 -c 'import aw_server'
+	pipenv run python3 -c 'import aw_server'
```
Comment (johan-bjareholt) [inline Makefile]: Having python installed does not necessarily mean that the user has pipenv installed, so if you want to change this default you should at least make sure that "pip3 install pipenv" is run before this.

Personally I prefer not using pipenv, so this also breaks my development workflow. But I guess for most people it's a good thing though.

Comment (ErikBjare) [reply Makefile]: It's to ensure that we always test with the exact-by-hash versions specified in the Pipfile.lock, so needed for reproducible builds.

**Comment by ErikBjare** [comment]
@johan-bjareholt This should be ready to merge now, can I get a final review?

File: .travis.yml [hunk 4]
```diff
@@ -5,20 +5,22 @@ python:
 #- "nightly"
 
 env:
- - TRAVIS_NODE_VERSION="8"
+ - SKIP_WEBUI=true  # TRAVIS_NODE_VERSION="8"
```
Comment (johan-bjareholt) [inline .travis.yml]: Nice!

**Pull request merged**